SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java

```diff
@@ -19,11 +23,26 @@ public ConferenceServiceImpl(ConferenceRepository conferenceRepository) {
 
     @Override
     public List<Conference> listConferences() {
-        return conferenceRepository.findAll();
+        List<Conference> conferences = conferenceRepository.findAll();
+        return conferences;
     }
 
     @Override
-    public Optional<Conference> findConferenceById(Long conferenceId) {
-        return conferenceRepository.findById(conferenceId);
+    public Conference findConferenceById(Long conferenceId) throws ConferenceNotFoundException {
+        Optional<Conference> optionalConference = conferenceRepository.findById(conferenceId);
+            return optionalConference.orElseThrow(() -> new ConferenceNotFoundException());
+
+    }
+    @Override
+    public Conference saveConference(Conference conference) throws ConferenceConflictException, InvalidConferenceFieldException {
+        if(conference.getId() == null || conference.getName() == null || conference.getDateTime() == null || conference.getCity() == null || conference.getDescription() == null || conference.getTopic() == null){
+            throw new InvalidConferenceFieldException();
+        }
+        if(conferenceRepository.existsById(conference.getId())) {
```

Comment:
Extract all these exception checks into separate methods.
Methods should be responsible for one thing only, this method is responsible for checking the existence of fields, the existence of the conference, whether the time is correct and saving the conference.

Reply:
So have 3 separate methods in `ConferenceServiceImpl` one for each check and then call all of them inside the `saveConference` method?